src/components/Calendar/Calendar.js

```diff
@@ -1,185 +1,142 @@
-				let cell = document.createElement('li');
-				let cellText = document.createTextNode('');
-				cell.appendChild(cellText);
-				calendarBody.appendChild(cell);
-			}
-		}
-
-		function next() {
-			year = month === 11 ? year + 1 : year;
-			month = (month + 1) % 12;
-			showCalendar(month, year);
-		}
-
-		function previous() {
-			year = month === 0 ? year - 1 : year;
-			month = month === 0 ? 11 : month - 1;
-			showCalendar(month, year);
-		}
-	};
-	componentDidMount() {
-		this.renderCalendar();
-	}
-
-	render() {
-		return (
-			<div className='events-calendar-container'>
-				<div className='calendar-content'>
-					<h2>Events Calendar</h2>
-					<div id='calendar'>
-						<div className='month'>
-							<ul>
-								<li id='prev'>&#10094;</li>
-								<li id='month'></li>
-								<li id='year'></li>
-								<li id='next'>&#10095;</li>
-							</ul>
-						</div>
-
-						<ul id='weekdays'>
-							<li>Su</li>
-							<li>Mo</li>
-							<li>Tu</li>
-							<li>We</li>
-							<li>Th</li>
-							<li>Fr</li>
-							<li>Sa</li>
-						</ul>
-
-						<ul id='days'></ul>
-					</div>
-				</div>
-			</div>
-		);
-	}
+  constructor(props) {
+    super(props);
+
+
+    const today = new Date();
+    this.state = {
+      dayInt: today.getDate(),
+      month: today.getMonth(),
+      year: today.getFullYear(),
+    };
+  }
+
+  handleClickNext = () => {
+    this.setState(({ month, year }) => {
+      const newYear = year = month === 11 ? year + 1 : year;
+      const newMonth = (month + 1) % 12;
+      
+      return {
+        month: newMonth,
+        year: newYear,
+      };
+    });
+  };
+
+  handleClickPrev = () => {
+    this.setState(({ month, year }) => {
+      const newYear = month === 0 ? year - 1 : year;
+      const newMonth = month === 0 ? 11 : month - 1;
+      
+      return {
+        month: newMonth,
+        year: newYear,
+      };
+    });
+  };
+
+  renderCalendar = () => {
+    const today = new Date();
+    const { events } = this.props;
+    const { dayInt, month, year } = this.state;
+    const firstDay = new Date(year, month).getDay();
+    const totalDays = new Date(year, month + 1, 0).getDate();
+
+    const calendarItems = [];
+
+    // render blank days first 
+    for (let x = 0; x < firstDay; x++) {
+      calendarItems.push(<li key={`blank-${x}`} />);
+    }
+
+    // render rest 
+    for (let day = 1; day <= totalDays; day++) {
+      const isActive = dayInt === day && month === today.getMonth() && year === today.getFullYear();
+
+      // we filter out all the vents on this day.
+      const eventsOnThisDay = events.filter((event) => {
+        const eventDate = event.date.split('-');
+        const eventDay = eventDate[2];
+        const eventMonth = eventDate[1];
+        const eventYear = eventDate[0];
+
+        return (
+          day === eventDay &&
+          (month + 1).toString() === eventMonth &&
+          year === eventYear
+        );
+      });
+
+      calendarItems.push(
```

Comment:
for each `for(let day...)` loop we add a new `li` to the array. 

again, each `li` will need a unique key because they are in an array.